src/back/spv/writer.rs

```diff
@@ -1680,8 +1686,43 @@ impl Writer {
                         MathOp::Other(inst, result_lookup_ty)
                     }
                     Mf::Normalize => MathOp::Single(spirv::GLOp::Normalize),
+                    Mf::FaceForward => MathOp::Triple(spirv::GLOp::FaceForward),
+                    Mf::Reflect => MathOp::Double(spirv::GLOp::Reflect),
+                    // exponent
+                    Mf::Exp => MathOp::Single(spirv::GLOp::Exp),
+                    Mf::Exp2 => MathOp::Single(spirv::GLOp::Exp2),
+                    Mf::Log => MathOp::Single(spirv::GLOp::Log),
+                    Mf::Log2 => MathOp::Single(spirv::GLOp::Log2),
                     Mf::Pow => MathOp::Double(spirv::GLOp::Pow),
                     // computational
+                    Mf::Sign => {
+                        let op = match self
+                            .get_type_inner(&ir_module.types, arg0_lookup_ty)
+                            .scalar_kind()
+                        {
+                            Some(crate::ScalarKind::Float) => spirv::GLOp::FSign,
+                            Some(crate::ScalarKind::Sint) => spirv::GLOp::SSign,
+                            other => unimplemented!("Unexpected sign({:?})", other),
+                        };
+                        MathOp::Single(op)
+                    }
+                    Mf::Fma => MathOp::Triple(spirv::GLOp::Fma),
+                    Mf::Mix => {
+                        let op = match self
+                            .get_type_inner(&ir_module.types, arg0_lookup_ty)
+                            .scalar_kind()
+                        {
+                            Some(crate::ScalarKind::Float) => spirv::GLOp::FMix,
+                            Some(crate::ScalarKind::Sint) => spirv::GLOp::IMix,
```

Comment:
There is a problem here. Looks like `IMix` is not in the extended instruction set - https://www.khronos.org/registry/spir-v/specs/1.0/GLSL.std.450.html
So for integers we'll have to actually encode the arithmetic expressions in here

Reply:
Done